core/src/sumeragi/main_loop.rs

```diff
@@ -1049,7 +1056,7 @@ trait ApplyBlockStrategy {
     fn before_update_hook(sumeragi: &mut Sumeragi);
 
     /// Operation to invoke in kura to store block.
-    fn kura_store_block(kura: &Kura, block: Arc<VersionedCommittedBlock>);
+    fn kura_store_block(kura: &Kura, block: VersionedCommittedBlock);
```

Comment:
Excellent, now we have tighter control over the lifetime of the blocks.

Reply:
yes, if we could also return a `&Block` instead of `Arc<Block>` from `kura` it would be a lot better. I had to go through some escapades with to return iterator over blocks because of this with the help of @Erigara